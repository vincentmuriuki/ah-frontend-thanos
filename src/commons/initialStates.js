const initialState = {
  articleReducer: {
    article: {
      id: 0,
      tag_list: [],
      slug: '',
      title: '',
      description: '',
      body: '',
      created_at: '',
      updated_at: '',
      image_url: '',
      audio_url: '',
      read_time: '',
      likes: 0,
      dislikes: 0,
      likeDislikeStatus: '',
      views_count: '',
      rating: 0,
      author: {
        id: 0,
        username: '',
        email: '',
      },
    },
    articles: [],
    errorMessage: '',
  },
  deleteArticleReducer: {},
  socialLoginReducer: { isLoggedIn: false },
  loginReducer: { errorMessage: '', successMessage: '', user_details: '' },
  userReducer: {
    freshUser: { email: '', password: '', username: '' },
  },
};

export const sampleArticle = {
  id: 1,
  tag_list: ['tag'],
  slug: 'title-sample',
  title: 'title sample',
  description: 'desc',
  body: 'body',
  created_at: '12-12-2018',
  updated_at: '12-12-2018',
  image_url: 'http://test.png',
  audio_url: 'http://test.mp3',
  read_time: '0 min',
  likes: 12,
  dislikes: 2,
  likeDislikeStatus: 'dislike',
  views_count: 0,
  rating: 0,
  author: {
    id: 2,
    username: 'janedoe',
    email: 'janedoe@gmail.com',
  },
};

export const sampleArticle2 = {
  ...sampleArticle,
  likeDislikeStatus: 'like',
};

export const initialStateWithSample = {
  article: sampleArticle,
};

export const initialStateWithSample2 = {
  article: sampleArticle2,
};

export default initialState;
