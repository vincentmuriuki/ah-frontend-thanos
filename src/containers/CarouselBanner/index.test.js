import React from 'react';
import { mount } from 'enzyme';
import { CarouselBanner } from '.';
import { sampleListOfArticles } from '../../commons/initialStates';

describe('<CarouselBanner /> container', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<CarouselBanner articles={sampleListOfArticles} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
