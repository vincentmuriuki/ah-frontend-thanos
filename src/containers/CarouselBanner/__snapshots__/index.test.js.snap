// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselBanner /> container should render 1`] = `
<CarouselBanner
  articles={
    Array [
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 1,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 2,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 3,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 4,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 5,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 6,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 7,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 8,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
      Object {
        "audio_url": "http://test.mp3",
        "author": Object {
          "email": "janedoe@gmail.com",
          "id": 2,
          "username": "janedoe",
        },
        "body": "body",
        "created_at": "12-12-2018",
        "description": "desc",
        "dislikes": 2,
        "id": 9,
        "image_url": "http://test.png",
        "likeDislikeStatus": "dislike",
        "likes": 12,
        "rating": 0,
        "read_time": "0 min",
        "slug": "title-sample",
        "tag_list": Array [
          "tag",
        ],
        "title": "title sample",
        "updated_at": "12-12-2018",
        "views_count": 0,
      },
    ]
  }
>
  <div
    className="carousel slide"
    data-ride="carousel"
    id="carouselExampleIndicators"
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        data-slide-to="0"
        data-target="#carouselExampleIndicators"
      />
      <li
        data-slide-to="1"
        data-target="#carouselExampleIndicators"
      />
      <li
        data-slide-to="2"
        data-target="#carouselExampleIndicators"
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="carousel-item carousel-image-item active"
        key="1"
      >
        <img
          alt="slide"
          className="d-block w-100"
          src="http://test.png"
        />
        <div
          className="carousel-caption d-none d-md-block carousel-article-desc"
        >
          <h5>
            <a
              href="article/1"
            >
              title sample
            </a>
          </h5>
          <p>
            desc
          </p>
        </div>
      </div>
      <div
        className="carousel-item carousel-image-item "
        key="2"
      >
        <img
          alt="slide"
          className="d-block w-100"
          src="http://test.png"
        />
        <div
          className="carousel-caption d-none d-md-block carousel-article-desc"
        >
          <h5>
            <a
              href="article/2"
            >
              title sample
            </a>
          </h5>
          <p>
            desc
          </p>
        </div>
      </div>
      <div
        className="carousel-item carousel-image-item "
        key="3"
      >
        <img
          alt="slide"
          className="d-block w-100"
          src="http://test.png"
        />
        <div
          className="carousel-caption d-none d-md-block carousel-article-desc"
        >
          <h5>
            <a
              href="article/3"
            >
              title sample
            </a>
          </h5>
          <p>
            desc
          </p>
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      data-slide="prev"
      href="#carouselExampleIndicators"
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      data-slide="next"
      href="#carouselExampleIndicators"
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
</CarouselBanner>
`;
